### PR TITLE
Improved error handeling and added some code comments.

### PR DESCRIPTION
**Pull request by a-breid-ericsson**
**Fixes issues when adding subscription in EI front end.**

- fixed bug name validation failed, name was only validated based on the last character, thus the name could be "My in%vali¤Name" and still pass.
- Improved error responses to user when entering invalid or no information.
- Added some code comments.

### CODE REVIEW

File: src/main/resources/static/js/subscription.js [hunk 1]
```diff
@@ -739,85 +739,95 @@ jQuery(document).ready(function () {
...
-            }
-        */
-
-        if (vm.subscription()[0].subscriptionName() == "") {
-            window.logMessages("Error: SubscriptionName field must have a value");
-            return;
-        }
+        // Validate notificationType field
         if (vm.subscription()[0].notificationType() == null) {
             window.logMessages("Error: notificationType field must boolean a value");
```
Comment (e-pettersson-ericsson) [inline src/main/resources/static/js/subscription.js]: This log does not make sense. Maybe have "Error: notificationType must have a value"?

File: src/main/resources/templates/subscription.html [hunk 2]
```diff
@@ -82,7 +82,7 @@ <h3 id="subData">Subscription Data</h3>
                         <h3 class="modal-title text-center" id="formHeader">Subscription Form</h3>
                     </div>
                 </div>
-                <div class="modal-body form" align="left">
+                <div class="modal-body form" allign="left">
```
Comment (e-pettersson-ericsson) [inline src/main/resources/templates/subscription.html]: Misspelled? Should be "align"

File: src/main/resources/templates/subscription.html [hunk 3]
```diff
@@ -234,12 +256,14 @@ <h3 class="modal-title text-center" id="formHeader">Subscription Form</h3>
                                                optionsText: $data.repeat(),
                                                value: $data.repeat,
                                                optionsCaption: 'Choose...'"></select>
+                                    <font class="hiddenField" id="repeatNotSet" color="red" size="2">> Repeat must be set</font>
```
Comment (e-pettersson-ericsson) [inline src/main/resources/templates/subscription.html]: One too many '>'

File: src/main/resources/templates/subscription.html (hunk 2 above)
Comment (a-breid-ericsson) [reply src/main/resources/templates/subscription.html]: Should be, but was marked red in Vs code, and OP with allign, think it does not matter so maybe to remove it.

File: src/main/resources/templates/subscription.html (hunk 3 above)
Comment (a-breid-ericsson) [reply src/main/resources/templates/subscription.html]: No the > is part of the message :)
Will looke like this [Dropdown](in re dcolor)> Repeat must be set

File: src/main/resources/static/js/subscription.js (hunk 1 above)
Comment (a-breid-ericsson) [reply src/main/resources/static/js/subscription.js]: Maybe to remove those messages since it error now appear in at the field where the values was added.

**Comment by e-edling-ericsson** [comment]
I think it would be great to know what kind of errors you have fixed in the PR. I think i could give a better review with that information :) I could with that information download the code and try those things out and see how it looks like in the browser.

File: src/main/resources/static/js/subscription.js (hunk 1 above)
Comment (e-pettersson-ericsson) [reply src/main/resources/static/js/subscription.js]: I think it's fine to keep both (as long as the information is the same) to clearly highlight that something is wrong.

File: src/main/resources/static/css/style.css [hunk 4]
```diff
@@ -55,6 +55,9 @@
     height: 45px;
     animation: spin 1s cubic-bezier(.86,0,.07,1) infinite;
 }
+.hiddenField {
```
Comment (Christoffer-Cortes) [inline src/main/resources/static/css/style.css]: There is a built in class in Bootstrap called 'd-none' that could be used.

**Pull request merged**